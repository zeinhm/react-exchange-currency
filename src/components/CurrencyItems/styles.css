.currency-item {
  border: 1px solid #999;
  display: flex;
  justify-content: space-between;
  margin: 1rem 1rem 0;
}

.currency-item > div {
  padding: 0 .5rem;
  width: 100%;
}

.currency-item > div > header {
  display: flex;
  justify-content: space-between;
  margin: 1rem 0;
}

.currency-item > div > p {
  font-style: italic;
  margin: 1rem 0;
}

.currency-item > div > p:nth-child(2) {
  font-size: .865rem;
  font-weight: bold;
}

.currency-item > div > p:nth-child(3) {
  font-size: .865rem;
}

.currency-item > button {
  border: none;
  border-left: 1px solid #999;
}