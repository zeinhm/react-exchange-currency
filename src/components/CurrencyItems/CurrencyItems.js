import React from 'react'
import { thousand } from '../../utils/format'
import './styles.css'

export default function CurrencyItems(props) {
  return (
    props.currencies.map((i, key) => (
      <div className='currency-item' key={key}>
        <div>
          <header>
            <p>{i.code}</p>
            <p>{thousand((i.rates * props.amount).toFixed(2))}</p>
          </header>
          <p>{i.name}</p>
          <p>1 USD = {i.code} {i.rates.toFixed(2)}</p>
        </div>
        <button onClick={()=> props.deleteCurrency(i.code)}>
          (-)
        </button>
      </div>
    ))
  )
}
