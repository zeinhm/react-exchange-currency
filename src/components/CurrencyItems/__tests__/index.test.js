import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import CurrencyItems from '../CurrencyItems';

jest.mock('../../../utils/format', () => ({ thousand: jest.fn() }));

describe('src/components/CurrencyItems', () => {
  test('render', () => {
    const shallow = new ShallowRenderer();
    const props = {
      amount: 100,
      code: 'test',
      currencies: [{ code: 'test', rates: 100}],
      deleteCurrency: () => {}
    }
    const tree = shallow.render(<CurrencyItems {...props} />);
    tree[0].props.children[1].props.onClick();
    expect(tree).toMatchSnapshot();
  });
});
