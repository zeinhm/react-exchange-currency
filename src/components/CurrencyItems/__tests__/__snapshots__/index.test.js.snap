// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/CurrencyItems render 1`] = `
Array [
  <div
    className="currency-item"
  >
    <div>
      <header>
        <p>
          test
        </p>
        <p />
      </header>
      <p />
      <p>
        1 USD = 
        test
         
        100.00
      </p>
    </div>
    <button
      onClick={[Function]}
    >
      (-)
    </button>
  </div>,
]
`;
